src/main/kotlin/io/github/gradlenexus/publishplugin/NexusPublishPlugin.kt

```diff
@@ -102,8 +102,8 @@ class NexusPublishPlugin : Plugin<Project> {
             stagingRepositoryId.convention(registryService.map { it.registry[repository.get().name].stagingRepositoryId })
         }
         extension.repositories.all {
-            username.convention(rootProject.providers.gradleProperty("${name}Username").forUseAtConfigurationTimeCompat())
-            password.convention(rootProject.providers.gradleProperty("${name}Password").forUseAtConfigurationTimeCompat())
+            username.convention(rootProject.provider { rootProject.findProperty("${name}Username") as? String })
+            password.convention(rootProject.provider { rootProject.findProperty("${name}Password") as? String })
```

Comment:
Soooo, good ol' [Murphy](https://en.wikipedia.org/wiki/Murphy%27s_law) has hit us. @szpak do you remember these conversations:

> I wonder, how many people could use the other cases mentioned in javadoc?
https://github.com/gradle-nexus/publish-plugin/pull/242#discussion_r1258921238

> ... and if someone does use it, and want it, they'll hopefully tell us :)
https://github.com/gradle-nexus/publish-plugin/pull/251#discussion_r1270942973

Yes, [E2E tests are using exactly those esoteric features](https://github.com/gradle-nexus/nexus-publish-e2e-minimal/blob/ac03ac44e67c28f59d065af252c5234ab96e60bb/build.gradle#L33)...

```gradle
project.ext[propertyNameToOverride] = project.getProperty("$e2ePropertyNameToOverride")
```

To re-test the rest of the #251 PR, for now I reverted the use of `gradleProperty` change in 8e96e5eb3c6cb1d3a00cc9d9c254443ced17d7c7

Reply:
I propose to merge this PR without these two lines of change, and revisit it separately, because of the potential changes needed in E2E tests.

Reply:
Sound good for me. Maybe also some other external projects use that "esoteric feature" :-).